src/infra/core/src/repos/metadata_chain_visitor.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::hash::Hash;
+
+use bitflags::bitflags;
+use kamu_core::{AppendError, HashedMetadataBlockRef};
+use opendatafabric::{MetadataBlock, MetadataEvent, Multihash};
+
+///////////////////////////////////////////////////////////////////////////////
+
+bitflags! {
+    #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+    pub struct MetadataBlockTypeFlags: u32 {
```

Comment:
Note: unlike the block below, this one will not signal if it's no longer complete. Some comments that suggest updating the masks whenever we add new event types could be nice.

Or better, can we generate these bit masks from ODF makefiles side ?

Reply:
@zaychenko-sergei , in general, with `impl From<&MetadataBlock> for MetadataBlockTypeFlags`, we definitely won't miss changes, but I agree with you -- it's better to move the Flags to the ODF land.

Do you mind if this is done in a follow-up PR?

Reply:
PR: https://github.com/open-data-fabric/open-data-fabric/pull/85